core/src/components/datetime/datetime.tsx

```diff
@@ -2021,7 +2056,11 @@ export class Datetime implements ComponentInterface {
                   flipRtl
                 ></ion-icon>
               </ion-button>
-              <ion-button aria-label="Next month" disabled={nextMonthDisabled} onClick={() => this.nextMonth()}>
+              <ion-button
+                aria-label="Next month"
+                disabled={nextMonthDisabled || disabled}
```

Comment:
We should be updating the `nextMonthDisabled` constant with the additional check for disabled.